src/osd/ReplicatedPG.cc

```diff
@@ -10457,8 +10463,10 @@ void ReplicatedPG::hit_set_persist()
   ::encode(ctx->new_obs.oi, boi);
 
   ctx->op_t->append(oid, 0, bl.length(), bl, 0);
-  setattr_maybe_cache(ctx->obc, ctx, ctx->op_t, OI_ATTR, boi);
-  setattr_maybe_cache(ctx->obc, ctx, ctx->op_t, SS_ATTR, bss);
+  map <string, bufferlist> attrs;
+  attrs[OI_ATTR] = boi;
+  attrs[SS_ATTR] = bss;
```

Comment:
these are marginally more efficient with attrs[OI_ATTR].claim(boi);
since the buffer ref counts don't need to be modified